File: nomad-samples/common/py/guest_registration/account/participant_panel_query.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json
import requests

# @param {string} AUTH_TOKEN - The authentication token

def participant_panel_query(AUTH_TOKEN: str, ID: str, API) -> dict:
    if not AUTH_TOKEN:
        raise Exception("Authentication token not found")

    if API == "portal":
        API_URL = f"{PORTAL_API_URL}/user-session/{ID}"
    else:
        API_URL = f"{PORTAL_API_URL}/admin/user-session/{ID}"

    HEADERS = {
        "Content-Type": "application/json",
      	"Authorization": "Bearer " + AUTH_TOKEN
    }
    
    try:
        RESPONSE = requests.get(API_URL, headers=HEADERS)
        if not RESPONSE.ok:
            raise Exception()

        return json.loads(RESPONSE.text)
    except:
      raise Exception("Participant panel query filed")

File: nomad-samples/common/js/guest-registration/guest/participant-panel-query.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js"; 

export default async function participantPanelQuery(AUTH_TOKEN, ID, API) 
{
    if (API === "portalApi") {
        API = `${prjConstants.PORTAL_API_URL}/user-session/${ID}`;
    }
    else
    {
        API = `${prjConstants.ADMIN_API_URL}/admin/user-session/${ID}`;
    }

		// Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);
  
    // Make the request
    const RESPONSE = await fetch(API, {
        method: "GET",
        headers: HEADERS
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        // Get the JSON from the response
        const INFO = await RESPONSE.json();

        return INFO;
    }
		
  	// There was an error
    await apiExceptionHandler(RESPONSE, "Participant panel query failed");
}